v2/inc/c_logging/log_context.h

```diff
@@ -28,6 +28,7 @@ extern "C" {
 
 #define LOG_MAX_STACK_DATA_SIZE                 4096
 #define LOG_MAX_STACK_PROPERTY_VALUE_PAIR_COUNT 64
+#define LOG_MAX_WCHAR_STRING_LENGTH 512
```

Comment:
>512 [](http://example.com/codeflow?start=36&length=3)

Just saying that in the world of our logging this size might be not enough.

I've encountered strings > 4K while doing certificates, simply due to the amount of information needed there.

What are the implications of making this 4k, let's say?

Reply:
changed to use 4k here.

In log_context_property_type_wchar_t_ptr.c line 127, allocated sizeof LOG_MAX_WCHAR_STRING_LENGTH to get the initial data size but not actually use the buffer and free right after get the data size in line 130. I was wondering if there would be some overhead if allocated more memory then needed here.

Reply:
Errr... logging absolutely cannot use any memory allocation.

Reply:
Also: doesn't seem changed (I still see it as 512)

Reply:
Also: seems to be defined in 2 places, which is the source of truth?

LOG_MAX_WCHAR_STRING_LENGTH in log_context.h:33 and in log_context_property_type_wchar_t_ptr.c:17

Reply:
LOG_MAX_WCHAR_STRING_LENGTH has been used in both 2 files, move the defination to log_context_property_type_wchar_t_ptr.h line 14 with value 4096